src/aish/terminal/pty/bash_rc_wrapper.sh

```
@@ -60,9 +65,61 @@ __aish_emit_session_ready() {
+
+__aish_load_command_metadata_from_fd() {
+    local metadata_line=""
+    local assignments=""
+
+    __aish_reset_command_metadata
+
+    if [[ ! "$__AISH_COMMAND_METADATA_FD" =~ ^[0-9]+$ ]]; then
+        return 1
+    fi
+
+    IFS= read -r -t 0.001 metadata_line <&${__AISH_COMMAND_METADATA_FD} || return 1
```

Shellcheck (0.11.0)

[info] Double quote to prevent globbing and word splitting.

(SC2086)